File: learn/learn-algo/leetcode/997/go/997.go
package main

func findJudge(n int, trust [][]int) int {
	var tIn = make([]int, n+1)
	var tOut = make([]int, n+1)
	res := -1

	for _, v := range trust {
		a := v[0]
		b := v[1]
		tIn[a]++
		tOut[b]++

	}

	for i := 1; i <= n; i++ {
		if tIn[i] == 0 && tOut[i] == n-1 {
			if res != -1 {
				return -1
			} else {
				res = i
			}
		}
	}
	return res
}

File: learn/learn-algo/leetcode/997/c++/997.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <numeric>
#include <math.h>
#include <unordered_set>
#include <unordered_map>
#include <map>
#include <set>

using namespace std;

//class Solution {
//public:
//    int findJudge(int n, vector<vector<int>>& trust) {
//        map<int, int> tMap;
//        for(int i = 1; i <= n; ++i) {
//            tMap.insert({i,1});
//        }
//
//        for(int i = 1; i <= n; i++) {
//            auto t = trust[i];
//            tMap[t[1]]++;
//        }
//
//        for(int i = 0; i < trust.size(); i++) {
//            auto t = trust[i];
//            tMap.erase(t[0]);
//        }
//
//        int res = -1;
//
//
//        for(auto t : tMap) {
//            if(t.second == n) {
//                res = t.first;
//            }
//        }
//        return res;
//    }
//};


class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        map<int, int> tMap;

        for(int i = 1; i <= n; ++i) {
            tMap.insert({i,1});
        }

        for(int i = 0; i < trust.size(); i++) {
            auto t = trust[i];
            tMap[t[1]]++;
        }

        for(int i = 0; i < trust.size(); i++) {
            auto t = trust[i];
            tMap.erase(t[0]);
        }

        int res = -1;


        for(auto t : tMap) {
            if(t.second == n-1) {
                res = t.first;
            }
        }
        return res;
    }
};